test/common/setInterval.spec.ts

```diff
@@ -60,10 +60,15 @@ describe('setInterval', function() {
     }, null, null, 'unit-test');
   });
 
-  it('should not cancel the task after invoke the setInterval callback', (done) => {
+  it('setInterval should not update task count to trigger onHasTask', (done: Function) => {
```

Comment:
I don't think this is right. Why should `setInterval ` be exempt from tasks?

Reply:
@mhevery , before this PR, I believe `setInterval` was exempt from tasks but still not correctly `updateTaskCount`.

https://github.com/angular/zone.js/blob/master/lib/zone.ts#L808

if the task is `isPeriodic`, it will be reset to `scheduled` state and it will not `updateTaskCount` after `runTask`, I think this is the correct behavior.
so it should not `updateTaskCount` when `scheduleTask` and `cancelTask` either.  so it will not trigger `onHasTask` wrongly.